crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -247,6 +247,34 @@ impl SearchPaths {
                     .and_then(|venv| venv.site_packages_directories(system))?
             }
 
+            PythonPath::Resolve(target, origin) => {
+                tracing::debug!("Resolving Python at `{target}`");
+
+                let root = system
+                    // If given a file, assume it's a Python executable, e.g., `.venv/bin/python3`,
+                    // and search for a virtual environment in the root directory. Ideally, we'd
+                    // invoke the target to determine `sys.prefix` here, but that's more complicated
+                    // and may be deferred to uv.
+                    .is_file(target)
```

Comment:
Should we also test if the file has the executable bit set (`path_metadata` instead of `is_file`)

Reply:
Seems reasonable, but is a little more painful cross-platform for little gain.

Reply:
Let me know if you think it's worth pursuing regardless.